### PR TITLE
Expose stacktrace along with the exemption in executor

### PR DESCRIPTION
**Pull request by segaz2002**
**Issue**
At the moment, the executor does not reveal the actual source of issue when an exemption is encountered within coney's executor and also there is no straight priviledge to accession `__STACKTRACE__` or `System.stacktrace/0` outside of rescue.

**Solution**
convert the first parameter of `error_happend` callback to a tuple where the first element will be the exemption itself and the second element can be the stacktrace.

**Benefit**
- Makes error discovery faster
- Better debugging, happy developer.

### CODE REVIEW

File: lib/coney/consumer.ex [hunk 1]
```diff
@@ -1,9 +1,11 @@
 defmodule Coney.Consumer do
+  @type stacktrace :: []
+  @type error_tuple :: {exception :: struct(), stacktrace}
```
Comment (llxff) [inline lib/coney/consumer.ex]: let's add name for stacktrace :) `{exception :: struct(), stacktrace :: stacktrace()}`

**Review by llxff — COMMENTED** [review]
Thank you, @segaz2002! Looks great! Could you please update readme and changelog as well?

File: lib/coney/consumer_executor.ex [hunk 2]
```diff
@@ -9,7 +9,7 @@ defmodule Coney.ConsumerExecutor do
   rescue
     exception ->
       if function_exported?(consumer, :error_happened, 3) do
-        exception
+        {exception, System.stacktrace()}
```
Comment (llxff) [inline lib/coney/consumer_executor.ex]: Also I think, maybe let's do it in backwards-compatible manner? then I will bump only `minor` version

```elixir
cond do
  function_exported?(consumer, :error_happened, 3) ->
    exception
    |> consumer.error_happened(payload, meta)
    |> handle_result(task)
  function_exported?(consumer, :error_happened, 4) ->
    exception
    |> consumer.error_happened(System.stacktrace(), payload, meta)
    |> handle_result(task)
  true ->
    reject(task)
end
```

what do you think?

Comment (segaz2002) [reply lib/coney/consumer_executor.ex]: Perfect.

**Pull request closed without merging**